tower-batch/tests/ed25519.rs

```diff
@@ -127,5 +141,29 @@ async fn batch_flushes_on_max_latency() -> Result<()> {
     // Use a very high max_items and a short timeout to check that
     // flushing is happening based on hitting max_latency.
     let verifier = Batch::new(Ed25519Verifier::new(), 100, Duration::from_millis(500));
-    timeout(Duration::from_secs(1), sign_and_verify(verifier, 10)).await?
+    timeout(Duration::from_secs(1), sign_and_verify(verifier, 10, None)).await?
+}
+
+#[tokio::test]
+async fn fallback_verification() -> Result<()> {
+    zebra_test::init();
+
+    let verifier = Fallback::new(
+        // Because the Batch wrapper was changed to have a generic error type,
+        // we have to specify it explicitly to avoid an inference hole.
+        // This is pretty unergonomic -- from the error message it's very unintuitive
+        // that the right fix is to change `Batch::new` to `Batch::<_,_,Stuff>::new`.
```

Comment:
we probably want to revert the change that makes the error type generic. When I tried applying it to `tower-buffer` the `clone` bound was a big issue and there's no real way around it that I can think of yet. This same issue is going to apply here so the generic fails to even accomplish the goal I had in mind which was being compatible with `eyre::Report` (I cant make eyre::Report clone...).

Alternatively, we can keep it clone while fixing this issue. When I was redoing these changes for `tower-buffer` eliza recommended keeping the `new` fn the way it is, where it doesn't expose the generic error parameter and instead sets the parameter to `Box<dyn Error>` on the service it returns by default. Then we'd add a second constructor `new2` or something that does expose the generic parameter so you can override it if you want. That way with the hard coded `new` fn and a defaulted generic parameter you should be able to completely ignore the fact that its generic.

Reply:
Yes, that was the concern: https://github.com/ZcashFoundation/zebra/pull/665/files#r455324847

Reply:
would you prefer to revert the generics PR or do the constructor fix? I can probably toss together the latter in a couple minutes

Reply:
So, the options are:

**1**: roll back the changes to make the error type generic
**2**: keep the error type generic (IIUC this requires a Clone bound (?)) but hardcode the error in the default constructor and add an additional code path that allows overriding the error.

IIUC the `Clone` problem comes from the fact that the buffer handles have to share access to errors generated by the worker, i.e., it's sort of fundamental to the problem and not something that can be designed away. If there's not a clear  solution for the `Clone` problem, it seems unlikely that there's another error type that we would be swapping in. In that case, it seems like we would be maintaining extra code to do generic errors, but not using it, which makes me think that (1) is the better option.  Does that seem right to you, or am I missing something?

Reply:
@yaahc Sorry, it took me a minute to write up the analysis above.  If that reasoning seems right to you, my preference would be to roll back the generic changes.

Reply:
https://github.com/ZcashFoundation/zebra/pull/666 I did it anyways, feel free to merge or close this PR, I don't have a preference on whether we keep the changes + this fix or revert them.

Reply:
> Does that seem right to you, or am I missing something?

I think this is correct. I don't think maintaining the clone stuff is a particularly big burden but there's literally no reason to keep it for us, so feel free to rip it out if you think that's an easier option. I already implemented the fix for the 2. option so I think they're both approximately equal cost short term, ill leave it up to you to decide which path to take for now.

Reply:
cf #668